### PR TITLE
Apply quality to each size and add to `wp_editor_set_quality` 

### PR DESCRIPTION
**Pull request by adamsilverstein**
Trac ticket: https://core.trac.wordpress.org/ticket/54648

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by Mte90** [comment]
any updates for this one?

**Comment by Mte90** [comment]
Just a ping

**Comment by mukeshpanchal27** [comment]
@adamsilverstein PR needs to update.

File: src/wp-includes/class-wp-image-editor.php [hunk 1]
```diff
@@ -260,11 +263,18 @@ public function set_quality( $quality = null ) {
...
 			 * @param string $mime_type Image mime type.
+			 * @param int[]  $size {
+			 *     Dimensions of the image.
+			 *
+			 *     @type int $width  The image width.
+			 *     @type int $height The image height.
+			 * }
 			 */
-			$quality = apply_filters( 'wp_editor_set_quality', $default_quality, $mime_type );
+			$quality = apply_filters( 'wp_editor_set_quality', $default_quality, $mime_type, $dims ? $dims : $this->size );
```
Comment (joemcgill) [inline src/wp-includes/class-wp-image-editor.php:277]: The array format that is being passed to the $dims parameter is slightly different from the format returned from $this->size.

$dims:

```php
array($width, $height);
```

$this->size:

```php
array( 'width' => $width, 'height' => $height );
```

To make these consistent, I would updated the new `$dims` array shape to be consistent with the preexisting shape of the array for the `$size` property. This way filters that are making use of this value will get a consistent array shape.

**Review by joemcgill — CHANGES_REQUESTED** [review]
Thanks @adamsilverstein. This looks really close to me. I tested today and can confirm that I was able to apply a filter that only changed the quality for a subset of images based on the dimensions of the image without affecting the original at all.

Example where only images over 1000px are being set to a low quality:

768x576
pasta-3-768x576

1024x768
pasta-3-1024x768

File: src/wp-includes/class-wp-image-editor.php (hunk 1 above)
Comment (adamsilverstein) [reply src/wp-includes/class-wp-image-editor.php:277]: @joemcgill I updated the PR to ensure the dimensions are always passed in a consistent way (matching $this->sizes, ie. using the width and height keys)

Comment (adamsilverstein) [reply src/wp-includes/class-wp-image-editor.php:277]: I also added a test to validate that the filter always receives this value correctly.

**Comment by adamsilverstein** [comment]
merged in https://core.trac.wordpress.org/changeset/59473

**Pull request closed without merging**